src/main/java/cloud/eppo/ConfigurationStore.java

```diff
@@ -1,105 +1,24 @@
-import java.util.stream.Collectors;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.jetbrains.annotations.NotNull;
 
 public class ConfigurationStore {
-  private static final Logger log = LoggerFactory.getLogger(ConfigurationStore.class);
-  private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule());
 
-  private Map<String, FlagConfig> flags;
-  private Map<String, BanditReference> banditReferences;
-  private Map<String, BanditParameters> banditParameters;
+  private volatile Configuration configuration;
 
-  public ConfigurationStore() {
-    flags = new ConcurrentHashMap<>();
-    banditReferences = new ConcurrentHashMap<>();
-    banditParameters = new ConcurrentHashMap<>();
-  }
-
-  public void setFlagsFromJsonString(String jsonString) {
-    FlagConfigResponse config;
-
-    try {
-      config = mapper.readValue(jsonString, FlagConfigResponse.class);
-    } catch (JsonProcessingException e) {
-      log.error("Unable to parse flag configuration response");
-      throw new RuntimeException(e);
-    }
-
-    if (config == null || config.getFlags() == null) {
-      log.warn("Flags missing in configuration response");
-      flags = new ConcurrentHashMap<>();
-      banditReferences = new ConcurrentHashMap<>();
+  public ConfigurationStore(final Configuration initialConfiguration) {
+    if (initialConfiguration != null) {
+      this.configuration = initialConfiguration;
     } else {
-      // TODO: atomic flags to prevent clobbering like android does
-      // Record that flags were set from a response so we don't later clobber them with a
-      // slow cache read
-      flags = new ConcurrentHashMap<>(config.getFlags());
-      banditReferences = new ConcurrentHashMap<>(config.getBanditReferences());
-      log.debug("Loaded {} flags from configuration response", flags.size());
-    }
-  }
-
-  public FlagConfig getFlag(String flagKey) {
-    if (flags == null) {
-      log.warn("Request for flag {} before flags have been loaded", flagKey);
-      return null;
-    } else if (flags.isEmpty()) {
-      log.warn("Request for flag {} with empty flags", flagKey);
-    }
-    return flags.get(flagKey);
-  }
-
-  public String banditKeyForVariation(String flagKey, String variationValue) {
-    // Note: In practice this double loop should be quite quick as the number of bandits and bandit
-    // variations will be small. Should this ever change, we can optimize things.
-    for (Map.Entry<String, BanditReference> banditEntry : banditReferences.entrySet()) {
-      BanditReference banditReference = banditEntry.getValue();
-      for (BanditFlagVariation banditFlagVariation : banditReference.getFlagVariations()) {
-        if (banditFlagVariation.getFlagKey().equals(flagKey)
-            && banditFlagVariation.getVariationValue().equals(variationValue)) {
-          return banditEntry.getKey();
-        }
-      }
+      configuration = Configuration.Builder.emptyConfig();
```

Comment:
super nit: having this method on `Configuration` could be cleaner:
```suggestion
      configuration = Configuration.empty();
```

Reply:
agreed.